File: Examples/VisualBasic/ProgrammersGuide/Working-with-Pages/GetVisioPagebyID.vb
Imports VisualBasic
Imports Aspose.Diagram
Imports System

Public Class GetVisioPagebyID
    Public Shared Sub Run()
        'ExStart:GetVisioPagebyID
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_VisioPages()

        ' Call the diagram constructor to load diagram from a VDX file
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))

        ' Set page id
        Dim pageid As Integer = 2
        ' Get page object by id
        Dim page2 As Page = diagram.Pages.GetPage(pageid)
        'ExEnd:GetVisioPagebyID
    End Sub
End Class

File: Examples/CSharp/ProgrammersGuide/Working-with-Pages/GetVisioPagebyID.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.ProgrammersGuide.Working_with_Pages
{
    public class GetVisioPagebyID
    {
        public static void Run()
        {
            // ExStart:GetVisioPagebyID
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_VisioPages();

            // Call the diagram constructor to load diagram from a VDX file
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");

            // Set page id
            int pageid = 2;
            // Get page object by id
            Page page2 = diagram.Pages.GetPage(pageid);
            // ExEnd:GetVisioPagebyID
        }
    }
}
